mezzaluna-feature-loader/src/loaded_features/iter.rs

```diff
@@ -0,0 +1,85 @@
+use core::iter::FusedIterator;
+use core::slice;
+use std::collections::hash_set;
+use std::path::{Path, PathBuf};
+
+use super::Feature;
+
+/// An iterator over the feature paths in a `LoadedFeatures`.
+///
+/// This struct is created by the [`iter`] method on [`LoadedFeatures`]. See its
+/// documentation for more.
+///
+/// [`iter`]: super::LoadedFeatures::iter
+/// [`LoadedFeatures`]: super::LoadedFeatures
+#[derive(Debug, Clone)]
+pub struct Iter<'a> {
+    pub(crate) inner: slice::Iter<'a, PathBuf>,
+}
+
+impl<'a> Iterator for Iter<'a> {
+    type Item = &'a Path;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let next = self.inner.next()?;
+        Some(&*next)
```

Comment:
I think all of these can do:

```
self.inner.next().as_deref()
```

Reply:
turns out no, this won't work. `Option::as_deref` requires `Option<T>` or `&Option<T>`, but here we have `Option<&T>` :(((((

Reply:
https://twitter.com/_lopopolo/status/1506513048827162626